pilot/pkg/bootstrap/validation.go

```diff
@@ -67,17 +69,26 @@ func (s *Server) initConfigValidation(args *PilotArgs) error {
 	})
 
 	if webhookConfigName := validationWebhookConfigName.Get(); webhookConfigName != "" {
-		iface, err := kube.NewInterfacesFromConfigFile(args.Config.KubeConfig)
-		if err != nil {
-			return err
-		}
-		client, err := iface.KubeClient()
-		if err != nil {
-			return err
-		}
-		dynamicInterface, err := iface.DynamicInterface()
-		if err != nil {
-			return err
+		var dynamicInterface dynamic.Interface
+		if s.kubeClient == nil || s.kubeConfig == nil {
+			iface, err := kube.NewInterfacesFromConfigFile(args.Config.KubeConfig)
+			if err != nil {
+				return err
+			}
+			client, err := iface.KubeClient()
+			if err != nil {
+				return err
+			}
+			s.kubeClient = client
+			dynamicInterface, err = iface.DynamicInterface()
+			if err != nil {
+				return err
+			}
+		} else {
+			dynamicInterface, err = dynamic.NewForConfig(s.kubeConfig)
+			if err != nil {
+				return err
+			}
```

Comment:
What is this change for ? Doesn’t seem related to dns

Reply:
It's related to testing and debugging the DNS code locally. I can move it to a separate PR, but I'm debugging all components locally and it's tricky to stash/unstash changes.